pkg/goversion/compat.go

```diff
@@ -2,28 +2,39 @@ package goversion
 
 import (
 	"fmt"
+	"os"
 )
 
 var (
 	MinSupportedVersionOfGoMajor = 1
 	MinSupportedVersionOfGoMinor = 15
 	MaxSupportedVersionOfGoMajor = 1
 	MaxSupportedVersionOfGoMinor = 17
-	goTooOldErr                  = fmt.Errorf("Version of Go is too old for this version of Delve (minimum supported version %d.%d, suppress this error with --check-go-version=false)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
+	goTooOldErr                  = fmt.Errorf("Version of Go may be too old to work with this version of Delve (minimum supported version %d.%d, suppress this error with --check-go-version=false)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
+	goTooOldWarn                 = fmt.Errorf("Warning: version of Go may be too old to work with this version of Delve (minimum supported version %d.%d)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
```

Comment:
Since we're erroring in this case, why not just say it's unsupported
```suggestion
	goTooOldErr                  = fmt.Errorf("Version of Go is not supported by this version of Delve (minimum supported version %d.%d, suppress this error with --check-go-version=false)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
	goTooOldWarn                 = fmt.Errorf("Warning: version of Go may be too old to work with this version of Delve (minimum supported version %d.%d)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
```

Reply:
I wanted to keep the wording consistent. The level of support doesn't change based on the flag value, only how we treat the check. If error/warning consistency is not critical, then we might as well just leave the original error as-is.

Reply:
It seems odd to error on a "may be".  The key point is that it's an unsupported version, and it's unsupported because it's too old.  WDYT:
```suggestion
	goTooOldErr                  = fmt.Errorf("Version of Go is not supported by this version of Delve (minimum supported version %d.%d, suppress this error with --check-go-version=false)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
	goTooOldWarn                 = fmt.Errorf("Warning: version of Go is not supported by this version of Delve and may not function correctly (minimum supported version %d.%d)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
```

Reply:
I revered to the original "too old" wording, but with a clarification about undefined behavior in the warning version. Also added Go version number to the message.